main.go

```diff
@@ -173,30 +169,27 @@ func (s *slackOutput) encodeMessage(fields map[string]interface{}) ([]byte, []st
 
 	// Assemble tags and message. This assumes that the only variability in the routes
 	// are the channel, user, and icon
-	tags := []string{}
-	var text string
+	messages := []slackMessage{}
 	for _, route := range routes {
-		// Last message wins (they should all be the same)
-		text = route.Message
+		// Make sure individual messages are not too long
+		if len(route.Message) > MaxMessageLength {
+			return nil, nil, fmt.Errorf("Exceeds max-length allowed by slack: %s", route.Message)
+		}
 
-		// Tag each message by channel, user, and icon
-		encTag, err := json.Marshal(slackTag{
+		messages = append(messages, slackMessage{
 			Channel:  route.Channel,
 			Username: route.User,
 			Icon:     route.Icon,
+			Text:     route.Message,
 		})
-		if err != nil {
-			return nil, nil, err
-		}
-		tags = append(tags, string(encTag))
 	}
 
-	// Make sure individual messages are not too long
-	if len(text) > MaxMessageLength {
-		return nil, nil, fmt.Errorf("Message length exceeds maximum length allowed by slack: %s", text)
+	encodedMsgs, err := json.Marshal(messages)
+	if err != nil {
+		return nil, nil, err
 	}
 
-	return []byte(text), tags, nil
+	return encodedMsgs, []string{"all"}, nil
```

Comment:
Is the `"all"` tag useful anymore? Hard to tell from my limited knowledge of what these tags were used for before but seems like a potential area for cleanup.

Reply:
In the notifications-consumer, no, tags aren't useful.  But they are used in both the analytics and alerts consumers.  Tags are a mechanism for sending one log-line to multiple batches.  For example, let's say a log-line needed to be sent to firehoseA, firehoseB, and firehoseC.  The encoder should return the list of tags `["firehoseA", "firehoseB", "firehoseC"]`.

As it turns out, batching in the notifications consumer is overkill, so now we just use one batch called "all".